### PR TITLE
[k8s] Add trust bundle for each module

### PR DESCRIPTION
**Pull request by dmolokanov**
In order to establish secure communication between module and iotedged each pod should contain volume with iotedged trust bundle attached.

### CODE REVIEW

**Comment by kkdawkins** [comment]
It looks fine, but since we are already behind on test debt, are there any UTs that you can write to cover this?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs [hunk 1]
```diff
@@ -459,130 +461,126 @@ async Task PruneServiceAccounts(List<string> accountNamesToPrune)
...
+                });
 
-            return new V1PodTemplateSpec();
+            var modulePodSpec = new V1PodSpec(containerList, volumes: volumeList, imagePullSecrets: imageSecret.OrDefault());
+            var objectMeta = new V1ObjectMeta(labels: podLabels, annotations: podAnnotations);
+            return new V1PodTemplateSpec(objectMeta, modulePodSpec);
         }
 
-        private VolumeOptions GetVolumesFromModule(IModule<AgentDocker.CombinedDockerConfig> moduleWithDockerConfig)
+        (List<V1Volume>, List<V1VolumeMount>, List<V1VolumeMount>) GetVolumesFromModule(IModule<AgentDocker.CombinedDockerConfig> moduleWithDockerConfig)
```
Comment (kkdawkins) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: Rather than put it back as a three tuple, why not create a VolumeOptions type?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/KubernetesModule.cs [hunk 2]
```diff
@@ -91,32 +96,33 @@ public class KubernetesModule : Module
...
-                if (this.enableServiceCallTracing)
-                {
-                    // enable tracing of k8s requests made by the client
-                    var loggerFactory = c.Resolve<ILoggerFactory>();
-                    ILogger logger = loggerFactory.CreateLogger(typeof(Kubernetes));
-                    ServiceClientTracing.IsEnabled = true;
-                    ServiceClientTracing.AddTracingInterceptor(new DebugTracer(logger));
-                }
+            builder.Register(
+                    c =>
```
Comment (kkdawkins) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/KubernetesModule.cs]: Why did you move these down to their own line? It increases the amount of nesting. Can you put these back?

Comment (kkdawkins) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/KubernetesModule.cs]: Same comment for the instances in this file.

Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/modules/KubernetesModule.cs]: Probably its IDE formatting that I need to configure

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs (hunk 1 above)
Comment (dmolokanov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/CrdWatcher.cs]: We don't use option here at all. Private type for private method seems like overkill or not?
